src/memray/commands/attach.py

```diff
@@ -297,6 +381,18 @@ def prepare_parser(self, parser: argparse.ArgumentParser) -> None:
 
     def run(self, args: argparse.Namespace, parser: argparse.ArgumentParser) -> None:
         verbose = args.verbose
+        mode = TrackingMode.ACTIVATE
+        duration = None
+        heap_size = None
+
+        if args.mode == "deactivate":
+            mode = TrackingMode.DEACTIVATE
+        elif args.heap_size:
+            mode = TrackingMode.UNTIL_HEAP_SIZE
+            heap_size = args.heap_size
+        elif args.duration:
+            mode = TrackingMode.FOR_DURATION
+            duration = args.duration
```

Comment:
We should have some extra checking that these different modes can't be missed. As it is, the user is allowed to run `memray attach --mode=deactivate --duration=10 --output=foo.bin` which doesn't make any sense. We should prevent `--deactivate-tracking` from being used with any other option except `--method`, and we should prevent `--heap-limit` and `--duration` from being used together.